### ISSUE TITLE
AssertionError in 2-processGraph step (by lib/verkko/scripts/fix_haplogaps.py)

### ISSUE THREAD

**Issue by chunlinxiao** [marbl/verkko#147]
Hi,

I was trying to test verkko  (v1.3.1) using hifi + ONT data, but encountered error as below:

"
......
mend <27362 >95052
mend >8441 >87541
mend <32264 >89329
Traceback (most recent call last):
  File "...miniconda3/lib/verkko/scripts/fix_haplogaps.py", line 203, in <module>
    assert edge[0] not in node_cuts
AssertionError
"
do you have any suggestion?

thanks in advance.

**Comment by skoren** [comment]
@maickrau could you take a look at this?

**Comment by maickrau** [comment]
Could you share the folder 2-processGraph?

**Comment by chunlinxiao** [comment]
the folder is too big (2.9G    2-processGraph.tar.gz) to be attached here (25 MB allowed) - any other way I can share with you (eg. googledrive )?

**Comment by skoren** [comment]
You can upload it to our FTP: https://canu.readthedocs.io/en/latest/faq.html#how-can-i-send-data-to-you

**Comment by chunlinxiao** [comment]
Thanks @skoren - I just uploaded 2-processGraph.tar.gz as suggested.

Many thanks for the help !

**Comment by skoren** [comment]
@maickrau this is in the globus share now.

**Comment by maickrau** [comment]
I forgot this uses the folder 1-buildGraph, could you also upload that folder.

**Comment by chunlinxiao** [comment]
@maickrau just uploaded to the same location as @skoren specified:

758M    1-buildGraph.tar.gz

thank both for looking into this !

**Comment by maickrau** [comment]
This crash should now be fixed in commit [508efb7](https://github.com/marbl/verkko/commit/508efb707f0ed3c5a8eb52a62cbe05381c835464). The assembly will probably be quite fragmented because the hifi coverage is pretty low.

**Comment by chunlinxiao** [comment]
thank you @maickrau - will it be a new release?

**Comment by chunlinxiao** [comment]
An update:  after replacing fix_haplogaps.py, the verkko workflow run successfully (all to the end) - indeed the assembly quite fragmented.


Also, for testing HiC option with verkko (bioconda verkko bioconda 1.3.1),  I included additional options "--hic1 hic_read1.fq.gz  --hic2 hic_read2.fq.gz" as suggested ("Verkko also supports hi-c phasing, reads should be provided with --hic1 and --hic2 options."),  but encountered an error saying:

Unknown option '--hic1'.
Unknown option '--hic2'.

not sure if anything I did was wroing?

thanks
